### ISSUE TITLE
Bug: Handshake fails using TLS webSocket handshake: http version was not 1.1

### ISSUE THREAD

**Issue by Lohann** [paritytech/jsonrpsee#1169]
I'm using [subxt](https://github.com/paritytech/subxt) for communication with some parachains, then I got unable to connect to Astar at `wss://rpc.astar.network:443`, when I attempt to connect to it, I get this error:
```
Error in the WebSocket handshake: http version was not 1.1
```

I think this is a bug because astar node is using http 1.1, and I'm able to connect to it using [tokio-tungstenite](https://github.com/snapview/tokio-tungstenite), so probably the issue is somewhere when decoding the packets.

Tracking down the error, it starts here:
https://github.com/paritytech/jsonrpsee/blob/v0.16.2/client/transport/src/ws/mod.rs#L298-L299
the function above uses [soketto](https://github.com/paritytech/soketto/tree/v0.7.1) calling this function:
https://github.com/paritytech/soketto/blob/v0.7.1/src/handshake/client.rs#L100-L115
the error happens when calling self.decode_response()? , which returns the error here:
https://github.com/paritytech/soketto/blob/v0.7.1/src/handshake/client.rs#L171-L173

**Comment by Lohann** [comment]
I see! thanks, this is how [tungstenite](https://github.com/snapview/tungstenite-rs/tree/v0.20.0) builds the handshake request:
https://github.com/snapview/tungstenite-rs/blob/v0.20.0/src/client.rs#L221-L240

Personally I think [fluent-uri](https://github.com/yescallop/fluent-uri-rs) is a better option over [http::uri](https://docs.rs/http/0.2.9/http/index.html#uris) and [url](https://docs.rs/url/latest/url/) for uri parsing.

**Comment by jsdw** [comment]
> I see! thanks, this is how [tungstenite](https://github.com/snapview/tungstenite-rs/tree/v0.20.0) builds the handshake request:
https://github.com/snapview/tungstenite-rs/blob/v0.20.0/src/client.rs#L221-L240

So, it does:

```rust
let host = authority
    .find('@')
    .map(|idx| authority.split_at(idx + 1).1)
    .unwrap_or_else(|| authority);
```

Where authority is eg `username:password@example.com:123`. It looks to me like it includes the port too then, but it may be that it's omitted from the Uri (as it can be if the default port it being used anyway).

We could omit it in thedefault case too probably, though it seems like the Astar node should handle this case since it's not an error to provide it in the default case.

**Comment by Lohann** [comment]
Seems each client handle that differently, for example the browser (chromium/brave) always omit the port when it is the default port (80 for http/ws or 443 for https/wss), even if you put it there explicitly, if is not the default port, always include it.

**Comment by niklasad1** [comment]
Fair enough, I opened an issue in the astar repo but we need to fix #1048 in order to omit the default ports which seems reasonable.

**Comment by niklasad1** [comment]
Closed by #1172
